### PR TITLE
More improvements to the test outcomes pages

### PR DESCRIPTION
**Pull request by pietroalbini**
This PR makes multiple changes to the test outcomes pages:

* The blurb about bare metal targets has been moved to the qualification plan, since it will be shared by multiple targets in the future, and on the report there is now a link to it.
* Unified the wording for tests always passing and tests only executed on hosts.
* Noted that tidy and linkchecker are only executed on hosts, rather than all targets.
* Switched to a table to display compiletest outcomes.
* Switched to a table to display cargo test outcomes.
* Moved the KP statement to the statement page, rather than duplicating it for each target.
* Removed the note around stages, as we don't expose them anymore.
* Changes CI to emit separate metrics for each crate.

### CODE REVIEW

**Comment by pietroalbini** [comment]
Will do another change in the PR, let's hold off merging it for now.

**Comment by pietroalbini** [comment]
bors try

Need to get metrics built by CI to see if it works correctly.

**Comment by pietroalbini** [comment]
Ok this should be ready for review.

**Comment by tshepang** [comment]
bors try

File: ferrocene/doc/qualification-plan/src/ci.rst [hunk 1]
```diff
@@ -116,3 +116,32 @@ All test results are compressed into a tarball which is included in the release
...
+
+To solve the issue, our approach is to create a new target based on the Rust
+target we need to test: this new "bare metal testing target" has the same
+configuration as the real target, with the only exception being enabling the
+operating system bindings for Linux. This new target won't be shipped to
+customers.
+
+The bare metal testing target allows us to execute the test suite on Linux
+(running on the hardware needed by the real target), side-stepping the
+requiremento to have an operative system.
```
Comment (tshepang) [inline ferrocene/doc/qualification-plan/src/ci.rst:142]:
```suggestion
requiremen to have an operating system.
```

File: ferrocene/doc/qualification-plan/src/ci.rst [hunk 2]
```diff
@@ -116,3 +116,32 @@ All test results are compressed into a tarball which is included in the release
...
+
+Since the only difference between the two targets is the implementation of the
+APIs in the ``std`` crate. Since the ``std`` crate is not shipped to customers
```
Comment (tshepang) [inline ferrocene/doc/qualification-plan/src/ci.rst:145]: there is something missing in this sentence

File: ferrocene/doc/qualification-plan/src/ci.rst [hunk 3]
```diff
@@ -116,3 +116,32 @@ All test results are compressed into a tarball which is included in the release
...
+customers.
+
+The bare metal testing target allows us to execute the test suite on Linux
+(running on the hardware needed by the real target), side-stepping the
+requirement to have an operating system.
+
+Since the only difference between the two targets is the implementation of the
+APIs in the ``std`` crate. and that crate is not shipped to customers for bare
+metal targets, we can conclude that the test results of the two targets are
+equivalent.
```
Comment (tshepang) [inline ferrocene/doc/qualification-plan/src/ci.rst:147]:
```suggestion
APIs in the ``std`` crate, and that crate is not shipped to customers for bare
metal targets, we can conclude that the test results of the two targets are
equivalent.
```

**Review by tshepang — APPROVED** [review]
bors merge

**Pull request merged**